src/tv2-common/helpers/graphics/internal/index.ts

```diff
@@ -44,132 +69,258 @@ export function CreateInternalGraphic(
+				lifespan: GetInfiniteModeForGraphic(
+					internalGraphic.engine,
+					internalGraphic.config,
+					internalGraphic.parsedCue,
+					internalGraphic.isStickyIdent
+				),
 				content: _.clone(content)
 			})
 		)
-	} else {
-		const piece = literal<IBlueprintPiece>({
-			externalId: partId,
-			name,
-			...(IsTargetingTLF(engine) || IsTargetingWall(engine)
-				? { enable: { start: 0 } }
-				: {
-						enable: {
-							...CreateTimingGraphic(config, parsedCue, !isStickyIdent)
-						}
-				  }),
-			outputLayerId,
-			sourceLayerId,
-			lifespan: GetInfiniteModeForGraphic(engine, config, parsedCue, isStickyIdent),
-			content: _.clone(content)
-		})
-		pieces.push(piece)
-
-		if (
-			sourceLayerId === SharedSourceLayers.PgmGraphicsIdentPersistent &&
-			(piece.lifespan === PieceLifespan.OutOnSegmentEnd || piece.lifespan === PieceLifespan.OutOnShowStyleEnd) &&
-			isStickyIdent
-		) {
-			// Special case for the ident. We want it to continue to exist in case the Live gets shown again, but we dont want the continuation showing in the ui.
-			// So we create the normal object on a hidden layer, and then clone it on another layer without content for the ui
-			pieces.push(
-				literal<IBlueprintPiece>({
-					...piece,
-					enable: { ...CreateTimingGraphic(config, parsedCue, true) }, // Allow default out for visual representation
-					sourceLayerId: SharedSourceLayers.PgmGraphicsIdent,
-					lifespan: PieceLifespan.WithinPart,
+	}
+}
+
+function createPiece(
+	internalGraphic: InternalGraphic,
+	pieces: IBlueprintPiece[],
+	content: IBlueprintPiece['content']
+): void {
+	const piece = literal<IBlueprintPiece>({
+		externalId: internalGraphic.partId ?? '',
+		name: internalGraphic.name,
+		...(IsTargetingTLF(internalGraphic.engine) || IsTargetingWall(internalGraphic.engine)
+			? { enable: { start: 0 } }
+			: {
+					enable: {
+						...CreateTimingGraphic(internalGraphic.config, internalGraphic.parsedCue, !internalGraphic.isStickyIdent)
+					}
+			  }),
+		outputLayerId: internalGraphic.outputLayerId,
+		sourceLayerId: internalGraphic.sourceLayerId,
+		lifespan: GetInfiniteModeForGraphic(
+			internalGraphic.engine,
+			internalGraphic.config,
+			internalGraphic.parsedCue,
+			internalGraphic.isStickyIdent
+		),
+		content: _.clone(content)
+	})
+	pieces.push(piece)
+
+	if (
+		internalGraphic.sourceLayerId === SharedSourceLayers.PgmGraphicsIdentPersistent &&
+		(piece.lifespan === PieceLifespan.OutOnSegmentEnd || piece.lifespan === PieceLifespan.OutOnShowStyleEnd) &&
+		internalGraphic.isStickyIdent
+	) {
+		// Special case for the ident. We want it to continue to exist in case the Live gets shown again, but we dont want the continuation showing in the ui.
+		// So we create the normal object on a hidden layer, and then clone it on another layer without content for the ui
+		pieces.push(CreateContinuationPieceForIdentPersistent(piece, internalGraphic))
+	}
+}
+
+export function CreateContinuationPieceForIdentPersistent(
```

Comment:
Is this a _continuation_ piece? I think this is a display-only piece that does nothing but indicate presence of the actual ident on the layer that is hidden. So perhaps something along the lines of `CreateIndicatorPieceForIdentPersistent`, `CreateMarkerPieceForIdentPersistent`, `CreateDummyPieceForIdentPersistent` would be better